### PR TITLE
Make it possible to get the appdata folder using the public API

### PR DESCRIPTION
**Pull request by CarlSchwan**
Can be helpful to port away from the IServerContainer::getAppDataDir

Needed for:

- [ ] https://github.com/nextcloud/social/pull/1415

### CODE REVIEW

File: lib/private/Server.php [hunk 1]
```diff
@@ -1439,6 +1439,8 @@ public function __construct($webRoot, \OC\Config $config) {
 
 		$this->registerAlias(IMetadataManager::class, MetadataManager::class);
 
+		$this->registerAlias(\OCP\Files\AppData\IAppDataFactory::class, \OC\Files\AppData\Factory::class);
```
Comment (ArtificialOwl) [inline lib/private/Server.php:1442]: <3

**Review by skjnldsv — APPROVED** [review]
Sure!

**Review by ArtificialOwl — APPROVED** [review]

**Comment by CarlSchwan** [comment]
ci failure unrelated

**Comment by juliushaertl** [comment]
I see this might be useful in some cases but in general apps can already obtain their own app data directory by injecting IAppData which should be the preferred way imo:

https://docs.nextcloud.com/server/latest/developer_manual/basics/storage/appdata.html
https://github.com/nextcloud/server/blob/0f33453610c46ef730192820017f8c1270b3096f/lib/private/AppFramework/DependencyInjection/DIContainer.php#L127-L129

**Pull request merged**